### PR TITLE
fix(styleguide): adds searchable icon filter

### PR DESCRIPTION
**Pull request by ronniechong**
This PR is adding a searchable filter for icons. The themable icon component is also refactored to use `useState`

filter

### CODE REVIEW

File: style-guide/src/components/design-nav/design-nav.js [hunk 1]
```diff
@@ -2,9 +2,13 @@ import React from 'react';
 import { Link } from 'gatsby';
 import styles from '../../styles/index.module.scss';
 
-const NavLink = (props) => (
+const NavLink = props => (
   <div className={styles.navLink}>
-    <Link to={props.to} exact={props.exact} activeClassName={styles.linkActive}>
```
Comment (ronniechong) [inline style-guide/src/components/design-nav/design-nav.js:7]: Gatsby 2 doesn't have `exact` props anymore

File: style-guide/src/components/design-nav/design-nav.js [hunk 2]
```diff
@@ -2,9 +2,13 @@ import React from 'react';
...
+    <Link
+      to={props.to}
+      partiallyActive={props.exact ? false : true}
```
Comment (ronniechong) [inline style-guide/src/components/design-nav/design-nav.js:9]: Using `exact` props to check for exact path. If we don't, refreshing the page will make the active link disappear.

File: style-guide/src/components/design-nav/design-nav.js [hunk 3]
```diff
@@ -13,15 +20,15 @@ const NavLink = (props) => (
...
+    <NavLink to="/">All Design Tokens</NavLink>
+    <NavLink to="/otkit/colors/">Colors</NavLink>
+    <NavLink to="/otkit/typography/">Typography</NavLink>
+    <NavLink to="/otkit/grids/">Grids</NavLink>
+    <NavLink to="/otkit/spacing/">Spacing</NavLink>
+    <NavLink to="/otkit/borders/">Borders</NavLink>
+    <NavLink to="/otkit/breakpoints/">Breakpoints</NavLink>
+    <NavLink to="/otkit/shadows/">Shadows</NavLink>
+    <NavLink to="/otkit/icons/">Icons</NavLink>
+    <NavLink to="/otkit/icons-theme/">Icons (theme)</NavLink>
```
Comment (ronniechong) [inline style-guide/src/components/design-nav/design-nav.js:28]: If there is no trailing slashes and if we refresh the page, the slash will automatically append to it and we won't be able to match the path and the styling will not apply.

We can use this if we want to keep non-trailing slash https://www.gatsbyjs.com/plugins/gatsby-plugin-remove-trailing-slashes/

File: style-guide/src/styles/otkit-icons.module.scss [hunk 4]
```diff
@@ -49,3 +49,12 @@ $width-main: 768px;
   padding: $spacing-xsmall;
   margin: $spacing-small 0;
 }
+
+.form {
```
Comment (ronniechong) [inline style-guide/src/styles/otkit-icons.module.scss:53]: Since we are revamping, I'll leave this styling simple for now

**Review by jakeclements — APPROVED** [review]
V.cool, thanks @ronniechong!

**Pull request merged**